base/libgit2/callbacks.jl

```diff
@@ -82,7 +82,8 @@ function credentials_callback(cred::Ptr{Ptr{Void}}, url_ptr::Cstring,
         end
 
         # first try ssh-agent if credentials support its usage
-        if creds[:usesshagent, credid] == "Y"
+        if creds[:usesshagent, credid] === nothing ||
```

Comment:
why is it getting used or passed around then? if it doesn't store anything, then all the logic is going to be wrong for it amd this is not going to be the only bug